nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Run/RunWorker.cs

```diff
@@ -202,17 +216,16 @@ await _apiHandler.IncrementMetric(new()
                                     Groups = previousDependency.Requirements.Single().Groups,
                                     Source = new RequirementSource()
                                     {
-                                        SourceUrl = analysisResult.UpdatedDependencies.Single(d => d.Name == dependency.Name).InfoUrl,
+                                        SourceUrl = analysisResult.UpdatedDependencies.FirstOrDefault(d => d.Name == dependency.Name)?.InfoUrl,
```

Comment:
Just for my own learning, why is this needed?

Reply:
In case it's not found, this will allow `null` to flow through instead of throwing.  It _should_ always be present, but if it's not and we can prevent the throw then we can at least continue with the update and PR creation.